dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/extension/GraceFulShutDown.java

```diff
@@ -0,0 +1,42 @@
+package org.apache.dubbo.config.spring.extension;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * this interface is used to enable grace shutdown
+ * any class implements this interface should be a spring bean
+ * @since 2.7.3
+ */
+public interface GraceFulShutDown {
```

Comment:
This interface is introduced to work and work only with spring framework. I am wondering: should we make it work for all scenarios? A quick thought is, should we make GracefulShutDown (or rename it to 'DubboLifecycle') an dubbo extension and let DubboShutdownHook to hold it?

Reply:
good idea,I will edit resolve it later!